File: Swift/isomorphic-strings.swift
class Solution {
    func isIsomorphic(_ s: String, _ t: String) -> Bool {
        if s.count != t.count {
            return false
        }

        var stDict = [Character: Character]()
        var tsDict = [Character: Character]()
        let sChars = [Character](s)
        let tChars = [Character](t)

        for i in 0..<s.count {
            let sCurrent = sChars[i]
            let tCurrent = tChars[i]

            if stDict[sCurrent] == nil && tsDict[tCurrent] == nil {
                stDict[sCurrent] = tCurrent
                tsDict[tCurrent] = sCurrent
            } else if stDict[sCurrent] != tCurrent || tsDict[tCurrent] != sCurrent {
                return false
            }
        }

        return true
    }
}

File: Python/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        stDict = dict()
        tsDict = dict()
        sChars = list(s)
        tChars = list(t)

        for i in range(len(s)):
            sCurrent = sChars[i]
            tCurrent = tChars[i]

            if (sCurrent not in stDict) and (tCurrent not in tsDict):
                stDict[sCurrent] = tCurrent
                tsDict[tCurrent] = sCurrent
            elif stDict[sCurrent] != tCurrent or tsDict[tCurrent] != sCurrent:
                return False

        return True
